### PR TITLE
Backport Mbedtls 2.1 - Added buffer_size and response_size options for ssl-server2.

### PR DESCRIPTION
**Pull request by AndrzejKurek**
This is a backport of #1140.

### CODE REVIEW

File: tests/ssl-opt.sh [hunk 1]
```diff
@@ -3675,70 +3964,662 @@ run_test    "Large packet TLS 1.2 BlockCipher larger MAC" \
...
+            -c "Read from server: 16384 bytes read"
+
+run_test    "Large server packet TLS 1.2 AEAD shorter tag" \
+            "$P_SRV response_size=16384" \
+            "$P_CLI force_version=tls1_2 \
+             force_ciphersuite=TLS-RSA-WITH-AES-256-CCM-8" \
+            0 \
+            -c "Read from server: 16384 bytes read"
+
+# Tests of asynchronous private key support in SSL
```
Comment (gilles-peskine-arm) [inline tests/ssl-opt.sh]: This feature doesn't exist in 2.1.

**Comment by AndrzejKurek** [comment]
@gilles-peskine-arm fixed.

**Comment by sbutcher-arm** [comment]
This PR unfortunately breaks the SSLv3 tests. If you do the following:
```
scripts/config.pl set MBEDTLS_SSL_PROTO_SSL3
make
tests/ssl-opt.sh -f 'Large server packet SSLv3 BlockCipher'
```
You get this:
```
Large server packet SSLv3 BlockCipher .................................. FAIL
  ! pattern 'Read from server: 16384 bytes read' MUST be present in the Client output
  ! outputs saved to o-XXX-1.log
------------------------------------------------------------------------
FAILED (0 / 1 tests (0 skipped))
```

I don't know if this is the test, or an issue with the library, but I'm removing the 'ready for merge' label, and marking it as 'needs work'.

**Comment by AndrzejKurek** [comment]
Updated and resolved the conflict. @k-stachowiak, @gilles-peskine-arm - please re-review.

**Pull request merged**